faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -440,25 +440,26 @@ private HttpRequest makeHttpRequest(String httpMethod, String path, Optional<Jso
 
     // If a query timeout has been given for the current request,
     // override the one from the Connection if any
-    Optional<Duration> queryTimeout = requestQueryTimeout.or(() -> defaultQueryTimeout);
+    // or use a default value
+    Duration queryTimeout = requestQueryTimeout.or(() -> defaultQueryTimeout).orElse(DEFAULT_REQUEST_TIMEOUT);
 
     Optional<Long> lastTxnTime = (getLastTxnTime() > 0) ? Optional.of(getLastTxnTime()) : Optional.empty();
 
     HttpRequest.Builder requestBuilder =
       HttpRequest.newBuilder()
         .uri(requestUri)
         .version(httpVersion)
-        .timeout(Duration.ofMillis(DEFAULT_REQUEST_TIMEOUT_MS))
```

Comment:
Does users have any ability to influence their request timeout? If not, that's an issue. Users may expect the query to take 2 minutes but the client will always timeout first. I believe unifying this timeouts would make for a simpler API.

Reply:
Sure, I can unify those :+1:

Reply:
WDYT about this https://github.com/fauna/faunadb-jvm/pull/298/commits/b2b6d7090758fd7b581629ec4b81d72ff500d96f

Reply:
Network latency is somewhat unpredictable. Those 10ms are not guaranteed to make a difference. I'd keep it simple just pass the query timeout  given that users have the ability to configure it from the driver's API if needed.